WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -0,0 +1,335 @@
+        if splitViewControllerIsHorizontallyCompact {
+            return
+        }
+
+        blogDetailsViewController?.showDetailView(for: .stats)
+    }
+
+    private func blogDetailsViewController(for blog: Blog) -> BlogDetailsViewController {
+        guard let blogDetailsViewController = blogDetailsViewController else {
+            let blogDetailsViewController = makeBlogDetailsViewController(for: blog)
+            self.blogDetailsViewController = blogDetailsViewController
+            return blogDetailsViewController
+        }
+
+        blogDetailsViewController.blog = blog
+        return blogDetailsViewController
+    }
+
+    private func makeBlogDetailsViewController(for blog: Blog) -> BlogDetailsViewController {
+        let blogDetailsViewController = BlogDetailsViewController(meScenePresenter: meScenePresenter)
+        blogDetailsViewController.blog = blog
+
+        return blogDetailsViewController
+    }
+
+    // MARK: - Model Changes
+
+    private func subscribeToModelChanges() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleDataModelChange(notification:)),
+            name: .NSManagedObjectContextObjectsDidChange,
+            object: ContextManager.shared.mainContext)
+    }
+
+    @objc
+    private func handleDataModelChange(notification: NSNotification) {
+        if let blog = blog {
+            handlePossibleDeletion(of: blog, notification: notification)
+        } else {
+            handlePossiblePrimaryBlogCreation(notification: notification)
+        }
+    }
+
+    // MARK: - Model Changes: Blog Deletion
+
+    /// This method takes care of figuring out if the selected blog was deleted, and to address any side effect
+    /// of the selected blog being deleted.
+    ///
+    private func handlePossibleDeletion(of selectedBlog: Blog, notification: NSNotification) {
+        guard let deletedObjects = notification.userInfo?[NSDeletedObjectsKey] as? Set<NSManagedObject>,
+           deletedObjects.contains(selectedBlog) else {
+            return
+        }
+
+        self.blog = nil
+    }
+
+    // MARK: - Model Changes: Blog Creation
+
+    /// This method ensures that the received notification includes inserted blogs.
+    /// It's useful because when we call `lastUsedOrFirstBlog()` a chain of calls that ends with:
+    ///
+    ///     `AccountService.defaultWordPressComAccount()`
+    ///     > `AccountService.accountWithUUID()`
+    ///     > `NSManagedObjectContext.executeFetchRequest(...)`.
+    ///
+    /// The issue is that `executeFetchRequest` updates the managed object context, thus triggering
+    /// a `NSManagedObjectContextObjectsDidChange` notification, which caused a neverending
+    /// loop in the observer in this VC.
+    ///
+    private func verifyThatBlogsWereInserted(in notification: NSNotification) -> Bool {
+        guard let insertedObjects = notification.userInfo?[NSInsertedObjectsKey] as? Set<NSManagedObject>,
```

Comment:
We could probably simplify like so:

```suggestion
        private func verifyThatBlogsWereInserted(in notification: NSNotification) -> Bool {
        (notification.userInfo?[NSInsertedObjectsKey] as? Set<NSManagedObject>)?.contains(where: { $0 as? Blog != nil }) == true
    }
```

Reply:
Thanks for the suggestion – I think I might leave it as-is, as I think it's perhaps a little clearer to read having it split out over a couple of lines 🙂